### PR TITLE
Various enhancements for tests, packaging and docs

### PR DESCRIPTION
**Pull request by philpep**
Hi,

Here is a long series adding and fixing various things related to tests, docs and packaging.
In particular this introduces flake8, docs, packaging tests and python 3.7 tests in travis config.

Feel free to pick individual patches if you don't want some of them (or ask me to drop them).

Thanks!

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,7 +1,27 @@
 [tox]
-envlist = py27, py35
+envlist = py27, py35, py37, flake8, docs, check-manifest
```
Comment (ccharles) [inline tox.ini]: Is `py36` excluded deliberately?

Comment (philpep) [reply tox.ini]: No, might be worth adding py36 as well.
On the other hand, testing all python versions might be overkill, tox support ``py3`` specifier which use version of the "python3" binary. On travis side we can test explicitly versions we want.

File: docs/conf.py [hunk 2]
```diff
@@ -57,9 +58,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = u'1.0.0'
+version = pkg_resources.get_distribution('anosql').version
```
Comment (nackjicholson) [inline docs/conf.py]: 😁 cool, didn't know about this!

File: tox.ini (hunk 1 above)
Comment (nackjicholson) [reply tox.ini]: Just keep testing py3.5 as well as py3.7 I suppose. We can skip py3.6, but I wan to know if something blows up in 3.5 or below because `anosql` is committed to maintaining support for it at this time (at least for this version). We would want to know if somebody accidentally used an `f""` string and broke py3.5.

Thanks for doing this, I hadn't used tox until I started helping with this project so I am by no means an expert at setting it up. I'm glad to have your help.

File: setup.py [hunk 3]
```diff
@@ -5,7 +5,8 @@
 
 setup(
     name='anosql',
-    version='1.0.1',
+    setup_requires=['setuptools_scm'],
```
[the remark this answers is not in the document]
Comment (philpep) [reply setup.py]: Sure, It's indeed not simple as it look like :)

https://github.com/pypa/setuptools_scm/ is maintained by the "Python Packaging Authority" and provide automatic version based on source control (git / hg etc) to provide a single version identifier.

For instance, on top of my head:
```
$ python setup.py --version
1.0.2.dev26+g362701f
```

This mean we are 26 commits on top of the tag 1.0.2 and the commit hash is 362701f (g is for "git").

So to release the 1.1.0 @honza will have to add a tag 1.1.0 and then ``python setup.py --version`` will return exactly "1.1.0".

TL;DR;

```
git tag X.Y.Z
git clean -fxd
python setup.py sdist
python setup.py bdist_wheel
twine upload dist/*
```
When running ``python setup.py sdist``, setuptools_scm will "freeze" the version in the tarbal "PKG-INFO" file.

But this will still require setuptools to pull setuptools_scm when installing. In particular this will fail with very old versions of setuptools (< 12 which is *very* old, but keep in mind that centos 7 still ship 0.9.6...).
To avoid this (and various issues related to old pip/setuptools versions), the state of the art is to also upload "wheel" packages along with the , which is a "pre-installed" package that doesn't require setuptools_scm and will install faster. (Btw when writting this I also added daebcde to build "universal" wheel)

Let me known if setuptools_scm is worth for you, if the release process appear to be too complex I can just drop this particular commit.

**Comment by philpep** [comment]
I rebased on top of master and added a flake8 fix for #38 on 0ae7499

**Comment by alecbrooks** [comment]
I know this was already approved, but nice work on the changes!

**Comment by nackjicholson** [comment]
@honza Bump! I think we would ideally like to merge these changes and release them as `1.0.2`

**Comment by philpep** [comment]
Hi @nackjicholson , I dropped the commit introducing setuptools_scm, so we can move forward on this, the release process is unchanged. I may resubmit it in a separate PR waiting a review from @honza

**Comment by honza** [comment]
Amazing! Thank you for your hard work on this

**Comment by honza** [comment]
I propose that we drop Python 2.x support. What do you think?
